### PR TITLE
6428 vizjson including widgets

### PR DESCRIPTION
**Pull request by juanignaciosl**
@viddo there was an ongoing discussion about what new attributes we should add here. If you already have a well known list I can do it now. Otherwise I'd better merge & deploy this, since vizjson3 is going to change a lot during the following weeks.

cc @javisantana @alonsogarciapablo @acanimal

### CODE REVIEW

**Comment by alonsogarciapablo** [comment]
As @viddo said (good catch!) we're missing some attributes that the new viz.json format requires.

The datasource groups some of the configuration options of layers of type `layergroup` and `namedmap` (which can be cleanup up)

```
   "datasource": {
            "user_name": "orgowner",
            "maps_api_template": "http://localhost.lan:8181/user/{user}",
            "stat_tag": "84ec6844-4b4b-11e5-9c1d-080027880ca6",
            "force_cors": true // Not sure if we need this???
            "template_name": "wadus"
          },
```
I'm not sure if `datasource` is the best name or we could find a better one. It's basically the configuration that CartoDB.js needs to create instances of that map in Windshaft...

We also have some attributes that are used in the blue side-bar.

```
          "title": "Time-series widget",
          "description": null,
          "user": {
            "fullname": "Emilio García",
            "avatar_url": "https://s3.amazonaws.com/com.cartodb.users-assets.production/production/piensaenpixel/assets/20150507074635Captura%20de%20pantalla%202015-05-07%20a%20la%28s%29%2009.46.28.png"
          }
```

**Comment by juanignaciosl** [comment]
Ok, this is the added information:

"datasource":{"user_name":"central-user-1","maps_api_template":"http://{user}.localhost.lan:8181","stat_tag":"7df5b4dc-bdab-11e5-95ec-080027880ca6","template_name":"tpl_7df5b4dc_bdab_11e5_95ec_080027880ca6"},"user":{"fullname":"Juan Ignacio S\u00e1nchez","avatar_url":"//example.com/avatars/avatar_mountains_blue.png"}}`

Is this ok?

**Comment by alonsogarciapablo** [comment]
I think so! I left a comment about `template_name`. Also, after we merge this, we should be able to flatten the layers in the viz.json (get rid of the "layergroup" and "namedmap" layers) which will make things a lot easier on your side!

**Comment by juanignaciosl** [comment]
@alonsogarciapablo : yes, it will always return a `template_name`. Ok, since you use it to differentiate between named an anonymous, I'll change it to be returned only in that case. Nevertheless I'd prefer an explicit check, but this is probably not the moment to discuss it :-)

**Comment by alonsogarciapablo** [comment]
Ok @juanignaciosl! Let's do it that way so that we don't have to change CartoDB.js/deep-insights.js/exmples right now. We can revisit it layer. cc: @javisantana

**Comment by juanignaciosl** [comment]
@ethervoid could you CR this, please? Build's failing because of unrelated errors.

@alonsogarciapablo, @viddo  : vizjson will not contain `template_name` for anonymous maps.

**Comment by juanignaciosl** [comment]
Retest this, please.

File: app/models/carto/widget.rb [hunk 1]
```diff
@@ -5,6 +5,10 @@ class Carto::Widget < ActiveRecord::Base
 
   validates :layer, :order, :type, :options, presence: true
 
+  def self.from_visualization_id(visualization_id)
+    Carto::Visualization.find(visualization_id).layers.map(&:widgets).flatten
```
Comment (ethervoid) [inline app/models/carto/widget.rb]: Just a question: This works if the visualization doesn't have layers? I know this should not happen but just in case

**Comment by ethervoid** [comment]
As far a is can read and document me about this is :+1:

File: app/models/carto/widget.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/models/carto/widget.rb]: Yes, because layers wouldn't be `nil` but `empty`.

**Pull request merged**